plugins/org.eclipse.n4js/src/org/eclipse/n4js/workspace/N4JSProjectConfigSnapshot.java

```diff
@@ -34,25 +35,28 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableList.Builder;
 import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
 
 /**
  * Extends Xtext's default {@link ProjectConfigSnapshot} by some additional attributes (e.g. project type).
  */
 public class N4JSProjectConfigSnapshot extends ProjectConfigSnapshot {
 
 	private final ProjectDescription projectDescription;
+	private final Map<String, String> packageNameToProjectIds;
 	private final boolean external;
 
 	/** Creates a new {@link N4JSProjectConfigSnapshot}. */
 	public N4JSProjectConfigSnapshot(ProjectDescription projectDescription, URI path,
 			boolean indexOnly, boolean generatorEnabled, Iterable<String> dependencies,
-			Iterable<? extends SourceFolderSnapshot> sourceFolders) {
+			Iterable<? extends SourceFolderSnapshot> sourceFolders, Map<String, String> packageNameToProjectIds) {
 
-		super(projectDescription.getName(), path,
+		super(projectDescription.getId(), path,
 				Collections.singleton(URIUtils.trimTrailingPathSeparator(path).appendSegment(N4JSGlobals.PACKAGE_JSON)),
 				indexOnly, generatorEnabled, dependencies, sourceFolders);
 
 		this.projectDescription = Objects.requireNonNull(projectDescription);
+		this.packageNameToProjectIds = packageNameToProjectIds;
```

Comment:
To guarantee immutability of the snapshot and for consistency with the super class, we should create a defensive copy here. I.e.:
```java
this.packageNameToProjectIds  = ImmutableMap.copyOf(packageNameToProjectIds);
```
(also the type of field `packageNameToProjectIds` should be ImmutableMap<..> (compare with super class))